src/ui/components/common/nav.rs

```diff
@@ -1,61 +1,155 @@
-use leptos::{component, view, IntoView};
+use crate::i18n::*;
+use leptos::{component, create_server_action, server, view, IntoView, ServerFnError};
 use leptos_icons::*;
 use leptos_router::*;
 
+#[server(LogoutFormFn, "/api")]
+pub async fn logout_form_fn() -> Result<(), ServerFnError> {
+  use crate::api::set_cookie_wrapper;
+  use leptos_actix::redirect;
+
+  redirect("/");
+
+  match set_cookie_wrapper("jwt", "").await {
+    Ok(o) => Ok(o),
+    Err(e) => Err(ServerFnError::ServerError(e.to_string())),
+  }
+}
+
 #[component]
 pub fn TopNav() -> impl IntoView {
+  let i18n = use_i18n();
+
+  let logout_form_action = create_server_action::<LogoutFormFn>();
+
   view! {
-    <div class="navbar bg-base-300">
+    <nav class="container navbar mx-auto">
       <div class="navbar-start">
-        <div class="dropdown">
-          <label tabindex="0" class="btn btn-ghost btn-circle">
-            <Icon icon=Icon::from(ChIcon::ChMenuHamburger) width="1.25rem" height="1.25rem"/>
-          </label>
-          <ul
-            tabindex="0"
-            class="menu menu-sm dropdown-content mt-3 z-[1] p-2 shadow bg-base-100 rounded-box w-52"
-          >
-            <li>
-              <a>"Homepage"</a>
-            </li>
-          </ul>
-        </div>
+        <ul class="menu menu-horizontal flex-nowrap">
+          <li>
+            <A href="/" class="text-xl whitespace-nowrap">
+              "Brand from env"
+            </A>
+          </li>
+          <li>
+            <A href="/communities" class="text-md">
+              {t!(i18n, nav_communities)}
+            </A>
+          </li>
+          <li>
+            <A href="/create_post" class="text-md">
+              {t!(i18n, nav_create_post)}
+            </A>
+          </li>
+          <li>
+            <A href="/create_community" class="text-md">
+              {t!(i18n, nav_create_community)}
+            </A>
+          </li>
+          <li>
+            <A href="//join-lemmy.org/donate">
```

Comment:
Starts with 2 slashes. I presume this is unintentional.

Reply:
without the double slash it get interpreted as a relative URL.

Reply:
I see. I think external urls use the lowercase a tag.